compiler/semantic/analyzer.go

```diff
@@ -49,6 +64,37 @@ func (a *analyzer) exitScope() {
 	a.scope = a.scope.parent
 }
 
+func (a *analyzer) buildFrom(op dag.Op) (dag.Op, error) {
+	switch op := op.(type) {
+	case *kernel.Reader, *dag.Lister, *dag.FileScan, *dag.HTTPScan, *dag.PoolScan, *dag.LakeMetaScan, *dag.PoolMetaScan, *dag.CommitMetaScan, *dag.DeleteScan:
+		return nil, nil
+	case *dag.Scope:
+		return a.buildFrom(op.Body[0])
+	case *dag.Fork:
+		return a.buildFrom(op.Paths[0][0])
```

Comment:
Nit: Alphabetize these so they match the order in compiler/ast/dag/op.go.
```suggestion
	case *dag.Fork:
		return a.buildFrom(op.Paths[0][0])
	case *dag.Scope:
		return a.buildFrom(op.Body[0])
```